aws/data_source_aws_service_discovery_dns_namespace.go

```diff
@@ -0,0 +1,100 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/servicediscovery"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func dataSourceServiceDiscoveryDnsNamespace() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceServiceDiscoveryDnsNamespaceRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"dns_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"DNS_PUBLIC",
```

Comment:
We should probably use the constants in the aws sdk in case these ever change in the sdk. You can find those here https://github.com/aws/aws-sdk-go/blob/master/service/servicediscovery/api.go#L6076

Reply:
@mbfrahry  thanks, its done.

$ make testacc TEST=./aws TESTARGS='-run=TestAccDataSourceAWSServiceDiscoveryDnsNamespace_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -parallel 20 -run=TestAccDataSourceAWSServiceDiscoveryDnsNamespace_basic -timeout 120m
=== RUN   TestAccDataSourceAWSServiceDiscoveryDnsNamespace_basic
=== PAUSE TestAccDataSourceAWSServiceDiscoveryDnsNamespace_basic
=== CONT  TestAccDataSourceAWSServiceDiscoveryDnsNamespace_basic
--- PASS: TestAccDataSourceAWSServiceDiscoveryDnsNamespace_basic (168.80s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	168.848s